servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -179,6 +154,34 @@ private void subscribeToEvents(boolean resubscribe) {
         }
     }
 
+    // This method is called eagerly, meaning the completable will be immediately subscribed to,
+    // so we don't need to do any Completable.defer business.
+    private Completable doClose(final boolean graceful) {
+        CompletableSource.Processor processor = Processors.newCompletableProcessor();
+        sequentialExecutor.execute(() -> {
```

Comment:
if there is an async hop or sequential execution on the same thread in different context this may not preserve AsyncContext correctly. `AsyncContext.wrapExecutor(..)` could be potentially be used or `AsyncContext.wrapRunnable()` internally if queue is involved.